content/blog/build-publish-containers-iac/index.md

```diff
@@ -0,0 +1,2252 @@
+{{< choosable language csharp >}}
+
+```csharp
+// Get registry info (creds and endpoint).
+var imageName = repo.RepositoryUrl;
+var registryInfo = repo.RegistryId.Apply(async (id) =>
+{
+    var creds = await GetCredentials.InvokeAsync(new GetCredentialsArgs { RegistryId = id });
+    var decodedData = Convert.FromBase64String(creds.AuthorizationToken);
+    var decoded = ASCIIEncoding.ASCII.GetString(decodedData);
+
+    var parts = decoded.Split(':');
+    if (parts.Length != 2)
+    {
+        throw new Exception("Invalid credentials");
+    }
+
+    return new ImageRegistry
+    {
+        Server = creds.ProxyEndpoint,
+        Username = parts[0],
+        Password = parts[1],
+    };
+});
+```
+
+{{< /choosable >}}
+
+#### Alternatively, Authenticate with ECR Credential Helper
+
+An alternative approach ECR supports is to [use the Docker credential helper from Amazon](https://aws.amazon.com/blogs/compute/authenticating-amazon-ecr-repositories-for-docker-cli-with-credential-helper/), which integrates with local IAM settings, adds smart caching, and removes the need for Docker to login. If you elect to go this route, simply leave out the username/password parts of the `docker.ImageRegistry`:
+
+{{< chooser language "javascript,typescript,python,go,csharp" / >}}
+
+{{< choosable language javascript >}}
+
+```javascript
+// Get registry info (creds and endpoint).
+var imageName = repo.repositoryUrl;
+var registryInfo = undefined; // use ECR credentials helper.
+```
+
+{{< /choosable >}}
+
+{{< choosable language typescript >}}
+
+```typescript
+// Get registry info (creds and endpoint).
+const imageName = repo.repositoryUrl;
+const registryInfo = undefined; // use ECR credentials helper.
+```
+
+{{< /choosable >}}
+
+{{< choosable language python >}}
+
+```python
+# Get registry info (creds and endpoint).
+image_name = repo.repository_url
+registry_info = None # use ECR credentials helper.
+```
+
+{{< /choosable >}}
+
+{{< choosable language go >}}
+
+```go
+// Get registry info (creds and endpoint).
+imageName := repo.RepositoryUrl
+registryInfo := docker.ImageRegistryArgs{} // use ECR credentials helper.
+```
+
+{{< /choosable >}}
+
+{{< choosable language csharp >}}
+
+```csharp
+// Get registry info (creds and endpoint).
+var imageName = repo.RepositoryUrl;
+// Omit registryInfo and use ECR credentials helper.
+```
+
+{{< /choosable >}}
+
+#### Learn More About ECR
+
+ECR offers many additional options not shown here. This includes configuring advanced IAM permissions, enabling image vulnerability scanning, managing the lifetime of images so that older, unused images are deleted based on configured policies, encrypting images, and more. For details on these advanced capabilities and more, refer to [Pulumi's ECR user guide](https://www.pulumi.com/docs/guides/crosswalk/aws/ecr/), [Pulumi's ECR API reference](https://www.pulumi.com/docs/reference/pkg/aws/ecr/repository/), or [Amazon's own product documentation](https://aws.amazon.com/ecr/).
```

Comment:
```suggestion
ECR offers many additional options not shown here. This includes configuring advanced IAM permissions, enabling image vulnerability scanning, managing the lifetime of images so that older, unused images are deleted based on configured policies, encrypting images, and more. For details on these advanced capabilities and more, refer to [Pulumi's ECR user guide](https://www.pulumi.com/docs/guides/crosswalk/aws/ecr/), [Pulumi's ECR API reference](https://www.pulumi.com/docs/reference/pkg/aws/ecr/repository/), or [Amazon's product documentation](https://aws.amazon.com/ecr/).
```